test/cli_shared.dart

```diff
@@ -657,6 +675,53 @@ void sharedTests(Future<TestProcess> runSass(Iterable<String> arguments)) {
       expect(sass.stdout, emitsDone);
       await sass.shouldExit(0);
     });
+    test("logs proper errors", () async {
+      var sass = await runSass(["--interactive"]);
+      sass.stdin.writeln("1 + 2;");
+      sass.stdin.writeln("max(2, 1 + blue)");
+      sass.stdin.writeln(r"1 + $x + 3");
+      sass.stdin.close();
+      expect(
+          sass.stdout,
+          emitsInOrder([
+            ">> 1 + 2;",
+            "        ^",
+            "Error: expected no more input.",
+            ">> max(2, 1 + blue)",
+            "          ^^^^^^^^",
+            'Error: Undefined operation "1 + blue".',
+            r">> 1 + $x + 3",
+            r"       ^^",
+            "Error: Undefined variable."
+          ]));
+      expect(sass.stdout, emitsDone);
+      await sass.shouldExit(0);
+    });
+    test("logs proper errors with color", () async {
+      var sass = await runSass(["--interactive", "--color"]);
+      sass.stdin.writeln("1 + 2;");
+      sass.stdin.writeln("max(2, 1 + blue)");
+      sass.stdin.writeln(r"1 + $x + 3");
+      sass.stdin.close();
+      expect(
+          sass.stdout,
+          emitsInOrder([
+            ">> 1 + 2;",
+            "\u001b[1F\u001b[8C\u001b[31m;",
+            "        ^",
+            "\u001b[0mError: expected no more input.",
+            ">> max(2, 1 + blue)",
+            "\u001b[1F\u001b[10C\u001b[31m1 + blue",
+            "          ^^^^^^^^",
+            '\u001b[0mError: Undefined operation "1 + blue".',
+            r">> 1 + $x + 3",
+            "\u001b[1F\u001b[7C\u001b[31m\$x",
+            r"       ^^",
+            "\u001b[0mError: Undefined variable."
+          ]));
+      expect(sass.stdout, emitsDone);
+      await sass.shouldExit(0);
+    });
```

Comment:
Test what happens when the user presses enter with a blank line. Right now it crashes due to the parse error problem I described above, but the correct answer is probably to just ignore the empty line.